File: Csharp/Job Matching#1.cs
// https://www.codewars.com/kata/56c22c5ae8b139416c00175d/train/csharp

using StriveObjects;
using System;

public class Strive
{
  public static bool Match(Candidate c, Job j)
  {
    if (c.MinSalary == null || j.MaxSalary == null)
    {
      throw new Exception();
    }
    
    return (c.MinSalary * 0.9 <= j.MaxSalary);    
  }
}

File: javascript/Job Matching#1.js
// https://www.codewars.com/kata/56c22c5ae8b139416c00175d/train/javascript

function match(candidate, job) {
    if (!candidate.minSalary || !job.maxSalary) throw "Invalid Job or Candidate";
     return job.maxSalary >= candidate.minSalary * 0.9;
   }